src/util/emitUtils.js

```diff
@@ -560,50 +560,50 @@ function getNewObjAsString(key, obj) {
  */
 function emitToClients(io, nsp, rooms, key, obj) {
   const namespace = io.of(nsp);
-  if (rooms && rooms.length) {
-    rooms.forEach((room) =>
-      namespace.to(room)
-    );
-    doEmit(namespace, key, obj);
-  } else if (key.startsWith('refocus.internal.realtime.sample')) {
-    tracker.trackEmit(obj.name, obj.updatedAt, 0); // wrap this around to check the key and include for only samples
-  } else if (key.startsWith('refocus.internal.realtime.bot.event')) {
-    doEmit(namespace, key, obj); // on an event emit to bots, there are no individual rooms specified
+  if (key.startsWith('refocus.internal.realtime.sample') && toggle.isFeatureEnabled('enableClientStats')) {
+    doEmitWithTracking(namespace, key, obj, rooms);
+  } else {
+    if (rooms && rooms.length) {
+      rooms.forEach((room) =>
+        namespace.to(room)
+      );
+      doEmit(namespace, key, obj);
+    }
   }
 }
 
 /**
- * Emit to the provided namespace, tracking stats if enabled.
+ * Emit to the provided namespace.
  * @param {Object}  nsp - the namespace to emit to
  * @param {String}  key - event type
  * @param {Object}  obj - event body
  */
 function doEmit(nsp, key, obj) {
+  const newObjectAsString = getNewObjAsString(key, obj); // { key: {new: obj }}
+  nsp.emit(key, newObjectAsString);
```

Comment:
We do still need to track the emit here, I was just saying we don't need to include the client count.

The idea for the tracking is that we log it at several different places so we can track the flow of a sample through the system:
upsert -> publish -> subscribe -> emit -> client receipt

Each phase is valuable because it allows us to compare the data there to the previous phase and get information about how many samples made it through and how long they took. 

The final phase for tracking the client receipt is behind a toggle because it's potentially expensive to do the extra callbacks and it's not strictly necessary - it's there just in case we need it, but normally won't be enabled. All the other phases will always be on.

The reason for including the client count at the emit phase is so that we can compare with the client count on receipt and see if we are dropping any events. So if we aren't tracking client receipt times, there's no need for the client count on emit.

But the emit phase still needs to be tracked so it can be compared with the other phases. It tells us whether the sample makes it through the filtering phase, and how long that takes.

So you should keep the call to trackEmit, but don't pass in the third numClients arg, and update the validation in trackEmit to make numClients optional.

Reply:
The client count does still need to be included when tracking callbacks, for the reason explained above.
trackEmit should be called in two places: from doEmit with key and obj, and from doEmitWithTracking with key, obj, and clientCount.

Reply:
That's what's currently happening, in doEmit we call trackEmit without the clients and below in doEmitWith tracking I have included sockets.length which is the number of clients emitted to.

Reply:
Oh, sorry, I got mixed up between tracker.trackEmit and pubsubStats.trackEmit - I saw that you removed pubsubStats.trackEmit from doEmitWithTracking and thought it was tracker.trackEmit. 

Looks good.